solidity-v1/dashboard/src/connectors/tally-injected.js

```diff
@@ -0,0 +1,76 @@
+import { AbstractConnector } from "./abstract-connector"
+import { WALLETS } from "../constants/constants"
+import BigNumber from "bignumber.js"
+import { UserRejectedConnectionRequestError } from "./utils"
+
+class TallyInjectedConnector extends AbstractConnector {
+  constructor() {
+    super(WALLETS.TALLY.name)    
+    this.provider = window.tally
+  }
```

Comment:
`TallyInjectedConnector` and `MetaMaskInjectedConnector` have basically the same code. The only difference is that it uses different provider and we set up `autoRefreshOnNetworkChange` to false for MetaMask. 

What I propose it to keep it as 'InjectedConnector' and extend both 'TallyInjectedConnector' and 'MetaMaskInjectedConnector' with it.

```suggestion
class InjectedConnector extends AbstractConnector {
  constructor(walletName, provider) {
    super(walletName)    
    this.provider = provider
  }
```

And then:
```
class TallyInjectedConnector extends InjectedConnector {
  constructor() {
    super(WALLETS.TALLY.name, window.tally)
  }
```

Note: The code was not tested.

Thoughts?

Reply:
https://github.com/keep-network/keep-core/pull/2744/commits/3248bcf4a251f24f0c6e09faa4402b92f67cb471